lualib/resty/events/worker.lua

```diff
@@ -187,61 +199,61 @@ communicate = function(premature)
     end
 
     if not exiting() then
-        assert(timer_at(random_delay(), function(premature)
-            communicate(premature)
-        end))
+        start_timer(self, random_delay())
     end
 end
 
-function _M.configure(opts)
-    assert(not _opts)
+function _M:init()
+    assert(self._opts)
 
-    _opts = opts
-
-    assert(timer_at(0, function(premature)
-        communicate(premature)
-    end))
+    start_timer(self, 0)
 
     return true
 end
 
 -- posts a new event
-local function post_event(source, event, data, spec)
-    local json, err
+local function post_event(self, source, event, data, spec)
+    local str, err
 
     EVENT_T.source = source
     EVENT_T.event = event
     EVENT_T.data = data
-    EVENT_T.pid = _worker_pid
+    EVENT_T.wid = _worker_id
 
     -- encode event info
-    json, err = encode(EVENT_T)
+    str, err = encode(EVENT_T)
 
-    if not json then
+    if not str then
         return nil, err
     end
 
     PAYLOAD_T.spec = spec or EMPTY_T
-    PAYLOAD_T.data = json
+    PAYLOAD_T.data = str
 
     -- encode spec info
-    json, err = encode(PAYLOAD_T)
+    str, err = encode(PAYLOAD_T)
 
-    if not json then
+    if not str then
         return nil, err
     end
 
-    local ok, err = _queue:push(json)
+    local ok, err = self._queue:push(str)
     if not ok then
         return nil, "failed to publish event: " .. err
     end
 
     return true
 end
 
-local function check_event(source, event)
-    if not _connected then
-        return nil, "not initialized yet"
+function _M:publish(target, source, event, data)
+    local ok, err
+
+    -- if not self._connected then
+    --     return nil, "not initialized yet"
+    -- end
+
+    if type(target) ~= "string" or target == "" then
+        return nil, "target is required"
     end
```

Comment:
Should we do `assert` here instead?

```suggestion
    assert(type(target) == "string" and target ~= "", "target is required")
```

Same below. I think `assert` is fine for input parameter validation. `nil, err` is good in other cases.